AttributeError: JointStatePublisher instance has no attribute 'source_update_cb'

Getting this error on Ubuntu 16.04 and ROS Kinetic:
`[ERROR] [1582565767.970762]: bad callback: <bound method JointStatePublisher.source_cb of <joint_state_publisher.JointStatePublisher instance at 0x7ff0c19dc5a8>>
Traceback (most recent call last):
  File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/topics.py", line 750, in _invoke_callback
    cb(msg)
  File "/opt/ros/kinetic/lib/python2.7/dist-packages/joint_state_publisher/__init__.py", line 202, in source_cb
    if self.source_update_cb is not None:
AttributeError: JointStatePublisher instance has no attribute 'source_update_cb'
`

Why `source_update_cb` is set to None aftrer subscribers are created? Shouldn't this variable be initialized before creating subscribers?

--- a/joint_state_publisher/src/joint_state_publisher/__init__.py
+++ b/joint_state_publisher/src/joint_state_publisher/__init__.py
@@ -160,15 +160,15 @@ class JointStatePublisher():
         else:
             self.init_urdf(robot)
 
-        source_list = get_param("source_list", [])
-        self.sources = []
-        for source in source_list:
-            self.sources.append(rospy.Subscriber(source, sensor_msgs.msg.JointState, self.source_cb))
-
         # The source_update_cb will be called at the end of self.source_cb.
         # The main purpose it to allow external observes (such as the
         # joint_state_publisher_gui) to be notified when things are updated.
         self.source_update_cb = None
+
+        source_list = get_param("source_list", [])
+        self.sources = []
+        for source in source_list:
+            self.sources.append(rospy.Subscriber(source, sensor_msgs.msg.JointState, self.source_cb))
 
         self.pub = rospy.Publisher('joint_states', sensor_msgs.msg.JointState, queue_size=5)
 

